Library/Homebrew/utils/curl.rb

```diff
@@ -0,0 +1,24 @@
+require "pathname"
+
+def curl_args(*args)
+  curl = Pathname.new ENV["HOMEBREW_CURL"]
+  curl = Pathname.new "/usr/bin/curl" unless curl.exist?
+  raise "#{curl} is not executable" unless curl.exist? && curl.executable?
+
+  flags = HOMEBREW_CURL_ARGS
+  flags.gsub!(" --progress-bar", "") if ARGV.verbose?
+
+  args = [curl, *flags.split, HOMEBREW_USER_AGENT_CURL, *args]
```

Comment:
This will need to be something like `[curl] + flags.split + [HOMEBREW_USER_AGENT_CURL] + args` to make Ruby 1.8.7 happy.